#! /bin/sh /usr/share/dpatch/dpatch-run
## fix-gcc4.3.dpatch by Jose Carlos Garcia Sogo <jsogo@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix compiling with gcc-4.3 (#455182)

@DPATCH@
diff -urNad gloox.git2~/src/connectionsocks5proxy.cpp gloox.git2/src/connectionsocks5proxy.cpp
--- gloox.git2~/src/connectionsocks5proxy.cpp	2008-03-05 22:49:54.000000000 +0100
+++ gloox.git2/src/connectionsocks5proxy.cpp	2008-03-05 22:50:26.000000000 +0100
@@ -38,6 +38,8 @@
 # include <sstream>
 #endif
 
+#include <cstdlib>
+
 namespace gloox
 {
 
diff -urNad gloox.git2~/src/tests/tag/tag_perf.cpp gloox.git2/src/tests/tag/tag_perf.cpp
--- gloox.git2~/src/tests/tag/tag_perf.cpp	2008-03-05 22:49:54.000000000 +0100
+++ gloox.git2/src/tests/tag/tag_perf.cpp	2008-03-05 22:50:26.000000000 +0100
@@ -9,6 +9,8 @@
 
 #include <sys/time.h>
 
+#include <cstdlib>
+
 static double divider = 1000000;
 static int num = 2500;
 static double t;
diff -urNad gloox.git2~/src/tests/zlib/zlib_perf.cpp gloox.git2/src/tests/zlib/zlib_perf.cpp
--- gloox.git2~/src/tests/zlib/zlib_perf.cpp	2008-03-05 22:49:54.000000000 +0100
+++ gloox.git2/src/tests/zlib/zlib_perf.cpp	2008-03-05 22:50:26.000000000 +0100
@@ -11,6 +11,8 @@
 #include <string>
 #include <sys/time.h>
 
+#include <cstdlib>
+
 #ifdef HAVE_ZLIB
 
 class ZlibTest : public CompressionDataHandler
diff -urNad gloox.git2~/src/tlsgnutlsbase.cpp gloox.git2/src/tlsgnutlsbase.cpp
--- gloox.git2~/src/tlsgnutlsbase.cpp	2008-03-05 22:49:54.000000000 +0100
+++ gloox.git2/src/tlsgnutlsbase.cpp	2008-03-05 22:50:26.000000000 +0100
@@ -18,6 +18,9 @@
 
 #include <errno.h>
 
+#include <cstdlib>
+#include <cstring>
+
 namespace gloox
 {
 
